Deprecate pedantic in linter rule set list (#3343)
dart-lang/site-www: src/tools/linter-rules.md

consider starting with a linter rule set,
which the following packages provide:

[lints]({{site.pub-pkg}}/lints)
[lints][]

> : The rule sets that the Dart team encourages using. 

Reviewer: Consider adding "for Dart command line and server programs. See `flutter_lints` for Flutter programs."

consider starting with a linter rule set,
which the following packages provide:

[lints][]

> : The rule sets that the Dart team encourages using
>   for command line and server applications. 
>   For Flutter programs, see [flutter_lints][].

Reviewer 2: This doesn't seem quite right, since we encourage these lints for _all_ Dart code, and command line and server aren't all the non-Flutter use cases for Dart. Also not sure we need a link to `flutter_lints`, since that's the next item.

Maybe:
```suggestion
: The rule sets that the Dart team encourages using.
  If you're writing Flutter code,
  use the `flutter_lints` rule set,
  which builds on `lints`.
```

Author: That makes a lot of sense with `flutter_lints` building on them. This seems nicer and less cluttered, thanks!

  partially determines the [score]({{site.pub}}/help/scoring) of
  packages uploaded to [pub.dev]({{site.pub}}).

[effective_dart][]
: A set of rules corresponding to the guidelines in [Effective Dart][].

[pedantic][] (_deprecated_)
: The deprecated set of rules previously used to match
  the rules used for all Google-internal Dart code.

>   Consider migrating to one of the [package:lints][lints] rule sets 
>   or [package:flutter_lints][flutter_lints] if you use Flutter.

Reviewer 2: I avoid using `package:blah` in the docs. How about:

```suggestion
  Consider migrating to one of the rule sets in
  the [lints][] or [flutter_lints][] package.
```
